src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,215 @@
+        Node<K, V> currentNode = table[calculateIndex(key, capacity)];
+        if (currentNode != null) {
+            if (isEqual(currentNode, newNode)) {
+                currentNode.value = value;
+                return;
+            }
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+                if (isEqual(currentNode, newNode)) {
+                    currentNode.value = value;
+                    return;
+                }
+            }
+            currentNode.next = newNode;
+            size++;
+            return;
+        }
+        table[calculateIndex(key, capacity)] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            if (table[0] == null) {
+                return null;
+            } else {
+                Node<K, V> current = table[0];
+                if (current.key == null) {
+                    return current.value;
+                }
+                while (current.next != null) {
+                    if (current.key == null) {
+                        return current.value;
+                    }
+                    current = current.next;
+                }
+            }
+        } else {
+            if (table[calculateIndex(key, capacity)] == null) {
+                return null;
+            } else {
+                Node<K, V> current = table[calculateIndex(key, capacity)];
+                Node<K, V> cover = new Node<>(key, null);
+                if (isEqual(current, cover)) {
+                    return current.value;
+                }
+                while (current.next != null) {
+                    current = current.next;
+                    if (isEqual(current, cover)) {
+                        return current.value;
+                    }
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public String toString() {
+        Node<K, V>[] nodes = new Node[size];
+        int index = 0;
+        for (int i = 0; i < capacity; i++) {
+            if (table[i] != null) {
+                nodes[index] = table[i];
+                index++;
+                if (table[i].next != null) {
+                    Node<K, V> current = table[i];
+                    while (current.next != null) {
+                        nodes[index] = current.next;
+                        index++;
+                        current = current.next;
+                    }
+                }
+            }
+        }
+        return Arrays.toString(nodes);
+    }
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            hash = getHash(key);
```

Comment:
You calculate it only once, why you need method?
```suggestion
            hash = key != null ? key.hashCode() : 0;
```